backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/services/FirstLevelChildUsageCreationNodeToolsProvider.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.syson.diagram.interconnection.view.services;
+
+import java.util.List;
+
+import org.eclipse.sirius.components.view.builder.IViewDiagramElementFinder;
+import org.eclipse.sirius.components.view.diagram.NodeTool;
+import org.eclipse.syson.diagram.interconnection.view.ILeveledNodeDescriptionNameGenerator;
+import org.eclipse.syson.diagram.interconnection.view.tools.PartUsageBindingConnectorAsUsageNodeToolProvider;
+import org.eclipse.syson.sysml.Element;
+import org.eclipse.syson.sysml.PartUsage;
+import org.eclipse.syson.util.IDescriptionNameGenerator;
+
+/**
+ * First level child creation node tool provider.
+ * The creation node tools provided by this class should be added to first level node descriptions only.
+ *
+ * @author Jerome Gout
+ */
+public class FirstLevelChildUsageCreationNodeToolsProvider implements ILeveledCreationNodeToolsProvider {
+    @Override
+    public List<NodeTool> getNodeTools(Element element, IDescriptionNameGenerator descriptionNameGenerator, IViewDiagramElementFinder cache) {
+        List<NodeTool> result = List.of();
+        if (descriptionNameGenerator instanceof ILeveledNodeDescriptionNameGenerator leveledNodeDescriptionNameGenerator) {
+            if (element instanceof PartUsage) {
```

Comment:
you could merged the two if statements in one

Reply:
Currently, `ILeveledCreationNodeToolsProvider.getNodeTools` is only used in the `casePartUsage` method of `InterconnectionViewNodeToolSectionSwitch`, so I could remove the inner test on the element type.
If in the future, we add another pair of node descriptions with the same structure (same domain type and one at the first level and the other one as a child) we need to separate the content of the `getNodeTools` according to the element type with something like:

```java
    public List<NodeTool> getNodeTools(Element element, IDescriptionNameGenerator descriptionNameGenerator, IViewDiagramElementFinder cache) {
        List<NodeTool> result = List.of();
        if (descriptionNameGenerator instanceof ILeveledNodeDescriptionNameGenerator leveledNodeDescriptionNameGenerator) {
            if (element instanceof PartUsage partUsage) {
                result = casePartUsage(partUsage, descriptionNameGenerator, cache, leveledNodeDescriptionNameGenerator);
            } else if (element instanceof  .... ) {
                result = ....
            }
        }
        return result;
    }
```
But I thought that was too much because the situation is not there yet.

Reply:
Ok thank you for the explanation